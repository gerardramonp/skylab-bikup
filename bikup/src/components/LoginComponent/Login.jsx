import React from 'react';
import { NavLink } from 'react-router-dom';

import './Login.scss';

function Login() {
    return (
        <>
            <div className="landing__header desktop">
                <img
                    className="logo"
                    src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
                    alt="logo"
                />
                <p className="appName">bikUP</p>
                <div className="flex-spacer"></div>
                <button className="login__button login__button--header">
                    Log In
                </button>
            </div>
            <div className="register__upper mobile">
                <NavLink to="/">
                    <img
                        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
                        alt="bikup logo"
                        className="register__logo"
                    />
                </NavLink>
                <NavLink to="/" className="register__bikup">
                    bikUP
                </NavLink>
            </div>
            <div className="login">
                <div className="login__container">
                    <h1>Log In</h1>

                    <form action="" className="login__form" method="POST">
                        <label htmlFor="email" className="input__label">
                            Email
                        </label>
                        <input
                            type="email"
                            className="login__email login__input"
                            name="email"
                        />
                        <label htmlFor="password" className="input__label">
                            Password
                        </label>
                        <input
                            type="password"
                            className="login__password login__input"
                            name="password"
                        />

                        <div className="form__bottom">
                            <NavLink to="/register">Create an account</NavLink>
                            <div className="flex-spacer"></div>
                            <button className="login__button" type="submit">
                                Submit
                            </button>
                        </div>
                    </form>

                    <div className="register__separator">
                        <div className="line"></div>
                        <span className="bold">Or</span>
                        <div className="line"></div>
                    </div>

                    <div className="login__external">
                        <button className="register__btn strava">
                            <div className="btn__logo">
                                <img
                                    src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/2f6af477333d585dfbfc19a14da9f857/strava-2.svg"
                                    alt="google logo"
                                />
                            </div>
                            <div className="btn__text">Login with Strava</div>
                        </button>

                        <button className="register__btn google">
                            <div className="btn__logo">
                                <img
                                    src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/99c6b9ab576aa9e324980db26e5fc7c0/google-logo.png"
                                    alt="google logo"
                                />
                            </div>
                            <div className="btn__text">Login with Google</div>
                        </button>
                    </div>
                </div>
            </div>
        </>
    );
}

export default Login;
