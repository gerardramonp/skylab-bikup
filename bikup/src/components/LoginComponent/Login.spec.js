import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';
import { BrowserRouter as Router } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('Login Component', () => {
	let tree = null;
	let compo = null;
	let mountedCompo = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<Login />
			</Router>
		);
	});

	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('should find button', () => {
		const page = shallow(<Login />);
		const button = page.find('.login__submitbutton');
		button.simulate('click', { preventDefault: () => {} });
		expect(button.length).toBe(1);
	});
});
