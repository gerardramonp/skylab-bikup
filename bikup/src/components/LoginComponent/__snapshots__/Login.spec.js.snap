// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match 1`] = `
Array [
  <div
    className="landing__header desktop"
  >
    <a
      aria-current="page"
      className="header__linkhome active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="logo"
        className="logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
      <p
        className="appName"
      >
        bikUP
      </p>
    </a>
    <div
      className="flex-spacer"
    />
  </div>,
  <div
    className="register__upper mobile"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="bikup logo"
        className="register__logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
    </a>
    <a
      aria-current="page"
      className="register__bikup active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      bikUP
    </a>
  </div>,
  <div
    className="login"
  >
    <div
      className="login__container"
    >
      <h1>
        Log In
      </h1>
      <form
        action=""
        className="login__form"
        method="POST"
      >
        <label
          className="input__label"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="login__email login__input"
          name="email"
          type="email"
        />
        <label
          className="input__label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="login__password login__input"
          name="password"
          type="password"
        />
        <div
          className="login__warning "
        />
        <div
          className="form__bottom"
        >
          <a
            aria-current={null}
            href="/register"
            onClick={[Function]}
          >
            Create an account
          </a>
          <div
            className="flex-spacer"
          />
          <button
            className="login__submitbutton"
            onClick={[Function]}
            onKeyUp={[Function]}
            type="submit"
          >
            <div
              className="loading hidden"
            >
              <img
                alt="loading..."
                className="loading__img"
                src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5753c458a8b552f891bb81/af512a8cb3c1285000d1191fdaaa670c/Spinner-1s-200px_(1).gif"
              />
              <p>
                Loading
              </p>
            </div>
            <div
              Account={true}
              Create={true}
              className="no-loading"
            >
              Log In
            </div>
          </button>
        </div>
      </form>
      <div
        className="register__separator"
      >
        <div
          className="line"
        />
        <span
          className="bold"
        >
          Or
        </span>
        <div
          className="line"
        />
      </div>
      <div
        className="login__external"
      >
        <button
          className="register__btn strava"
        >
          <a
            className="button__anchor button__anchor--strava"
            href="http://www.strava.com/oauth/authorize?client_id=37664&response_type=code&redirect_uri=http://localhost:3000/auth/strava&approval_prompt=force&scope=read_all,activity:read,profile:read_all"
          >
            <div
              className="btn__logo"
            >
              <img
                alt="google logo"
                src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/2f6af477333d585dfbfc19a14da9f857/strava-2.svg"
              />
            </div>
            <div
              className="btn__text"
            >
              Login with Strava
            </div>
          </a>
        </button>
        <button
          className="register__btn google"
        >
          <div
            className="btn__logo"
          >
            <img
              alt="google logo"
              src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53a18bf970328231db4f61/99c6b9ab576aa9e324980db26e5fc7c0/google-logo.png"
            />
          </div>
          <div
            className="btn__text"
          >
            Login with Google
          </div>
        </button>
      </div>
    </div>
  </div>,
]
`;
